### PR TITLE
Node VCX Wrapper resolves path correctly in OSX

### PR DESCRIPTION
**Pull request by Patrik-Stas**
Apart from resolving proper postfix for the library file, I am also proposing to change expected directory for OSX to `/usr/local/lib` which seems as a more proper place for this kind of library, and also in order to to stay consistent with indysdk wrapper. IndySDK Nodejs wrapper expects `libindy.dylib` to be located at `/usr/local/lib`, see https://github.com/hyperledger/indy-sdk/blob/master/wrappers/nodejs/README.md
I guess this change will break someone's stuff! Maybe we should have both `/usr/local/lib` and `/usr/lib` for OSX?

And I am not sure about linux, but seems like the indysdk nodejs wrapper expects it to be in `/usr/lib`. 

 Didn't test on Windows, so not sure if it really works there.

### CODE REVIEW

**Comment by Patrik-Stas** [comment]
Seems like I accidentally tried to merge to master myself.

**Comment by jovfer** [comment]
@Patrik-Stas Thank you for the contribution!
However, it cannot be accepted without a sign-off from you about intellectual property.
Please see https://github.com/probot/dco#how-it-works and instruction about how to fix it: https://github.com/hyperledger/indy-sdk/blob/master/docs/contributors/signing-commits.md#how-to-sign-previous-commits

Could you sign your commit (first one) and update the PR?

**Comment by Patrik-Stas** [comment]
@jovfer thank you, will do

**Comment by Patrik-Stas** [comment]
@jovfer Signed off.

File: vcx/wrappers/node/src/vcx.ts [hunk 1]
```diff
@@ -18,14 +19,21 @@ export class VCXRuntime {
...
   private _initializeBasepath = (): string => {
-    const library = 'libvcx.so' // TODO: FIXME provide better way to resolve library
+    const platform = os.platform()
+    const postfix = (platform === 'darwin') ? 'dylib' : (platform === 'win32') ? 'dll' : 'so'
+    const library = `libvcx.${postfix}`
+    const libDir = (platform === 'darwin')
+        ? '/usr/local/lib/'
+        : (platform === 'win32')
+            ? 'c:\\windows\\system32\\'
+            : '/usr/lib/'
```
Comment (Artemkaaas) [inline vcx/wrappers/node/src/vcx.ts]: I suggest using dictionaries for extensions and paths like:
`extention = {"darwin": ".dylib", "linux": ".so", "win32": ".dll"}`

File: vcx/wrappers/node/test/suite2/ffi.test.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ describe('vcxInit', () => {
 // these tests were created to only test that the ffi could be called with each function
 
 describe('Using the vcx ffi directly', () => {
-  const path = '/usr/lib/libvcx.so'
+  const path = '/usr/local/lib/libvcx.so'
   const run = new VCXRuntime({ basepath: path })
```
Comment (Artemkaaas) [inline vcx/wrappers/node/test/suite2/ffi.test.ts]: our CI runs vcx python tests on Ubuntu only. I believe we have to leave `/usr/lib` or to build it properly base on os.

File: vcx/wrappers/node/src/vcx.ts (hunk 1 above)
Comment (Patrik-Stas) [reply vcx/wrappers/node/src/vcx.ts]: changed

File: vcx/wrappers/node/test/suite2/ffi.test.ts (hunk 2 above)
Comment (Patrik-Stas) [reply vcx/wrappers/node/test/suite2/ffi.test.ts]: reverted

**Pull request merged**